docs/commit_practives.md

```diff
@@ -0,0 +1,112 @@
+
+## What are the benefits
+
+- Communicate the nature of changes to teammates, the public, and other stakeholders.
+- Make it easier for people to contribute to your projects, by allowing them to explore a more structured commit history.
+- Automatically generate CHANGELOGs.
+- Automatically determine a semantic version bump (based on the types of commits landed).
+- Trigger build and publish processes.
+
+## Commit format
+
+The commit message should be structured as follows:
+
+```txt
+<type>[optional scope][optional "!"]: <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+### Type
+
+The type of commit message is a required parameter and should be one of the following:
+
+- **feat**: A new feature
+- **fix**: A bug fix
+- **docs**: Documentation only changes
+- **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+- **refact**: A code change that neither fixes a bug nor adds a feature
+- **perf**: A code change that improves performance
+- **test**: Adding tests or correcting existing tests
+- **build**: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
+- **ci**: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
+- **chore**: Other changes that don't modify src or test files
+- **revert**: Reverts a previous commit
+
+### Optional scope
+
+The scope provides additional contextual information about the commit and is enclosed in brackets. It is optional and can be omitted if the commit does not apply to a specific component.
+
+If it does apply, the scope should not take more than 18 characters. If it takes more, that means the scope you are trying to describe is too detailed. Choose a higher level scope instead.
+
+A scope can include a name of a component and its subcomponent, separated by a slash. A subcomponent is usually the lowest level of component that is being modified.
+
+Example components: `UI`, `API`, `Models`, `Hooks`
+
+Example subcomponents: `tx modal`, `estimate`, `user`, `useFetch`
+
+A component and subcomponent should be separated by a slash `/`, e.g. `UI/tx modal`.
+
+If this commit introduces a component or a subcomponent for the first time, then it should not be mentined in the scope, but in the description.
+For example, if you are adding a new component `UI/tx modal`, then the scope should be omitted, and the commit message should look like this:
+
+```txt
+feat: add tx modal component
+```
+
+Then, when modifying this component, you must use the scope `UI/tx modal`.
+
+```txt
+fix(UI/tx modal): not closing on click
+```
+
+### Description
+
+The description is a brief summary of the changes made in the commit. It should be clear, concise, and written in the imperative mood.
+
+The type, scope and description can not exceed 50 charecters. Therefore, the description should be short and to the point.
```

Comment:
Are we talking about sum of chars along all three categories? That feels too restrictive, to be honest. I see no issue in large descriptions that would state multiple things like motivation for a change, what has been done to each subcomponent, some important things to keep in mind.